src/hidimstat/clustered_inference.py

```diff
@@ -255,17 +281,17 @@ def clustered_inference(
     train_index = _subsampling(n_samples, train_size, groups=groups, seed=seed)
 
     # Clustering
-    X, ward = _ward_clustering(X_init, ward, train_index)
+    X, ward = memory.cache(_ward_clustering)(X_init, ward, train_index)
 
     # Preprocessing
     X = StandardScaler().fit_transform(X)
```

Comment:
This might not be the preferred way to do it if we have categorical data in X. Maybe specifying `continuous_cols` would be useful. We also have the utils `conditional_sampling._check_data_type` that could be useful.

Reply:
See will be modified in the PR #156.